terraform/slurm_cluster/modules/_instance_template/main.tf

```diff
@@ -103,7 +107,7 @@ resource "google_compute_instance_template" "tpl" {
   dynamic "service_account" {
     for_each = var.service_account == null ? [] : [var.service_account]
     content {
-      email  = lookup(service_account.value, "email", null)
+      email  = "${data.google_project.default_sa.number}-compute@developer.gserviceaccount.com"
```

Comment:
It should be 

```hcl
email = lookup(service_account.value, "email", "${data.google_project.this.number}-compute@developer.gserviceaccount.com")
```

Reply:
Changed to `coalesce(var.service_account.email, "${data.google_project.this.number}-compute@developer.gserviceaccount.com")` per offline discussion.